sympy/vector/scalar.py

```diff
@@ -13,11 +13,17 @@ class BaseScalar(Symbol):
     """
 
     def __new__(cls, name, index, system, pretty_str, latex_str):
-        name = str(name)
-        pretty_str = str(pretty_str)
-        latex_str = str(latex_str)
         from sympy.vector.coordsysrect import CoordSysCartesian
-        obj = super(BaseScalar, cls).__new__(cls, name)
+        if isinstance(name, Symbol):
+            name = name.name
+        if isinstance(pretty_str, Symbol):
+            pretty_str = pretty_str.name
+        if isinstance(latex_str, Symbol):
+            latex_str = latex_str.name
+
+        index = _sympify(index)
+        system = _sympify(system)
+        obj = super(BaseScalar, cls).__new__(cls, Symbol(name), index, system, Symbol(pretty_str), Symbol(latex_str))
```

Comment:
With that suggestion, CoordSysCartesian's args would include the names given to its base vectors and scalars. As a result, we would not need to pass the string forms (l.e. the args Symbol(pretty_str), Symbol(latex_str) in the code above) during the construction of BaseScalar. They would be available from the corresponding system instance, which is already known to BaseScalar.